src/pocketmine/world/World.php

```diff
@@ -1754,7 +1754,11 @@ public function useItemOn(Vector3 $vector, Item &$item, int $face, ?Vector3 $cli
 			}
 		}
 
-
+		$tx = new BlockTransaction($this);
+		if(!$hand->place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player) or !$tx->apply()){
```

Comment:
This will write the blocks to the world before the event is called.

I did mention previously that the return of this function is not intended to signify anything with regards to events.

Reply:
You're right. Check last commit, I fix it.